File: go/01/part1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"path/filepath"
	"strconv"
)

func main() {
	path, _ := filepath.Abs("../input/01/question.txt")
	file, err := os.Open(path)
	if err != nil {
		panic(err)
	}
	defer file.Close()

	best_elf := 0
	current_elf := 0
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()

		if len(line) == 0 {
			if current_elf > best_elf {
				best_elf = current_elf
			}
			current_elf = 0
			continue
		}

		calories, _ := strconv.Atoi(line)
		current_elf += calories
	}

	if current_elf > best_elf {
		best_elf = current_elf
	}

	fmt.Println(best_elf)
}

File: zig/01/part1.zig
const std = @import("std");

pub fn main() !void {
    var file = try std.fs.cwd().openFile("../input/01/question.txt", .{});
    defer file.close();

    var reader = std.io.bufferedReader(file.reader());
    var stream = reader.reader();
    var buffer: [1024]u8 = undefined;

    var best: u32 = 0;
    var current: u32 = 0;

    while (try stream.readUntilDelimiterOrEof(&buffer, '\n')) |line| {
        const end = if (line[line.len-1] == '\r') line.len - 1 else line.len;
        const value = std.fmt.parseInt(u32, line[0..end], 10);

        if (value) |val| {
            current += val;
        } else |_| {
            if (current > best) {
                best = current;
            }

            current = 0;
        }
    }

    std.log.info("{d}", .{best});
} 
